src/Microsoft.AspNet.Mvc.TagHelpers/Internal/FileProviderGlobbingDirectory.cs

```diff
@@ -0,0 +1,96 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using Microsoft.AspNet.FileProviders;
+using Microsoft.Framework.FileSystemGlobbing.Abstractions;
+
+namespace Microsoft.AspNet.Mvc.TagHelpers.Internal
+{
+    public class FileProviderGlobbingDirectory : DirectoryInfoBase
+    {
+        private readonly IFileProvider _fileProvider;
+        private readonly IFileInfo _fileInfo;
+        private readonly FileProviderGlobbingDirectory _parent;
+
+        public FileProviderGlobbingDirectory(
+            IFileProvider fileProvider,
+            IFileInfo fileInfo,
+            FileProviderGlobbingDirectory parent)
+        {
+            _fileProvider = fileProvider;
+            _fileInfo = fileInfo;
+            _parent = parent;
+            if (_parent != null && !string.IsNullOrEmpty(_parent.RelativePath) && _fileInfo != null)
+            {
+                RelativePath = _parent.RelativePath + Path.DirectorySeparatorChar + _fileInfo.Name;
+            }
+            else if (_fileInfo != null)
+            {
+                RelativePath = _fileInfo.Name;
+            }
+            else
+            {
+                RelativePath = string.Empty;
+            }
+        }
+
+        public string RelativePath { get; private set; }
+
+        public override string FullName
+        {
+            get
+            {
+                if (_parent == null || string.IsNullOrEmpty(_parent.FullName))
+                {
+                    return Name;
```

Comment:
Not sure exactly what is the case here (add a comment).

perhaps should this be  `_parent.Name + Path.DirectorySeparatorChar + Name` perhaps with another if for the case where `_parent.Name` is `null`. Again not sure, since there is no comment